automation/repeated_test.sh

```diff
@@ -41,67 +44,78 @@ export TIMESTAMP=${TIMESTAMP:-1}
+usage: [NUM_TESTS=x]
+       [NEW_TESTS=file_name.json]
+       [TARGET_COMMIT_RANGE=a1b2c3d4] $0 [TEST_LANE] [--dry-run]
 
-    run tests repeatedly using the set of test files that have been changed or added since last merge commit
-    set NEW_TESTS to explicitly name the test files to run
+    run set of tests repeatedly using a json containing test names of tests that have been
+    changed or added since last merge commit
+    hint: set NEW_TESTS to explicitly name the json file containing test names to run
 
     options:
-        NUM_TESTS       how often the test lane is run, default is 5
-        NEW_TESTS       what set of tests to run, defaults to all test files added or changed since
-                        last merge commit
-        TARGET_COMMIT   the commit id to use when fetching the changed test files
-                        note: leaving TARGET_COMMIT empty only works if on a git branch different from main.
-                        If /clonerefs is at work you need to provide a target commit, as then the latest commit is a
-                        merge commit (resulting in no changes detected)
-        TEST_LANE       the kubevirtci provider to use, if not given, use latest stable one
+        NUM_TESTS           how often the test lane is run, default is 5
+        NEW_TESTS           the json file containing the (textual) names of the
+                            tests to run, according to what is shown in the
+                            junit.xml file
+        TARGET_COMMIT_RANGE the commit id to use when fetching the changed
+                            test files
+                            note: leaving TARGET_COMMIT_RANGE empty only works
+                            if on a git branch different from main.
+                            If /clonerefs is at work you need to provide a
+                            target commit, as then the latest commit is a
+                            merge commit (resulting in no changes detected)
+        TEST_LANE           the kubevirtci provider to use, if not given,
+                            use latest stable one
 
     examples:
 
-      1.    NEW_TESTS='tests/operator_test.go' ./automation/repeated_test.sh 'k8s-1.27'
+      1.    NEW_TESTS='/tmp/changed-tests.json' ./automation/repeated_test.sh 'k8s-1.27'
 
-            runs tests/operator_test.go x times on kubevirtci provider k8s-1.27
+            runs set of tests contained in /tmp/changed-tests.json x times
+            on kubevirtci provider k8s-1.27
 
-      2.    NEW_TESTS='tests/operator_test.go' ./automation/repeated_test.sh
+      2.    NEW_TESTS='/tmp/changed-tests.json' ./automation/repeated_test.sh
 
-            runs tests/operator_test.go x times on latest stable kubevirtci provider found
+            runs set of tests contained in /tmp/changed-tests.json x times
+            on latest stable kubevirtci provider found
 
 EOF
 }
 
+# see CANNIER PR at https://github.com/kubevirt/project-infra/pull/3930
```

Comment:
Do we depend on that PR?

Reply:
mentioning that PR was just meant as a reference, however that PR is not yet merged

Reply:
Yeah, makes sense - I'll elaborate on the comment